confluence-syntax-xhtml/src/main/java/org/xwiki/contrib/confluence/parser/xhtml/internal/wikimodel/PreformattedTagHandler.java

```diff
@@ -46,12 +48,16 @@ public PreformattedTagHandler()
     @Override
     protected void begin(TagContext context)
     {
-        setAccumulateContent(true);
+        WikiParameters wikiParameters = new WikiParameters();
+        wikiParameters = wikiParameters.setParameter("class", "code");
```

Comment:
I wonder if it would make sense to also take any potential parameters that are on the pre tag.